ext/intl/locale/locale_methods.c

```diff
@@ -125,7 +219,7 @@ static char* getPreferredTag(const char* gf_tag)
 	if(grOffset < 0) {
 		return NULL;
 	}
-	if( grOffset < LOC_PREFERRED_GRANDFATHERED_LEN ){
+	if( LOC_PREFERRED_GRANDFATHERED[grOffset] != NULL ){
```

Comment:
Previously, `LOC_PREFERRED_GRANDFATHERED` had only 6 values, and we relied on this to determine whether to return the preferred tag or the grandfathered tag. If the offset was less than 6, then we returned the preferred value.

Now, both arrays are the same size, and the indexes should always match up. If there's a `NULL` value in `LOC_PREFERRED_GRANDFATHERED` that matches the index of the grandfathered tag found in `LOC_GRANDFATHERED`, then we know it doesn't have a preferred value.